composer/datasets/ade20k.py

```diff
@@ -0,0 +1,336 @@
+import os
+from dataclasses import dataclass
+from math import ceil
+from typing import List, Optional, Tuple, Union
+
+import numpy as np
+import torch
+import torchvision.transforms.functional as TF
+import yahp as hp
+from PIL import Image
+from torch.utils.data import Dataset
+from torchvision import transforms
+
+from composer.core.types import Batch, Tensor
+from composer.datasets.hparams import DataloaderSpec, DatasetHparams
+from composer.utils import ddp
+
+
+class TransformationFn:
+    """Normalizes input data and removes the background class from target data if desired.
+
+    Args:
+        ignore_background (bool): Whether or not to ignore the background class when calculating the training loss.
+    """
+
+    def __init__(self, ignore_background: bool = True) -> None:
+        self.mean: Optional[Tensor] = None
+        self.std: Optional[Tensor] = None
+        self.ignore_background = ignore_background
+
+    def __call__(self, batch: Batch):
+        xs, ys = batch
+        assert isinstance(xs, Tensor)
+        assert isinstance(ys, Tensor)
+        device = xs.device
+
+        if self.mean is None:
+            self.mean = torch.tensor([0.485 * 255, 0.456 * 255, 0.406 * 255], device=device)
+            self.mean = self.mean.view(1, 3, 1, 1)
+        if self.std is None:
+            self.std = torch.tensor([0.229 * 255, 0.224 * 255, 0.225 * 255], device=device)
+            self.std = self.std.view(1, 3, 1, 1)
+
+        xs = xs.float()
+        xs = xs.sub_(self.mean).div_(self.std)
+        if self.ignore_background:
+            ys = ys.sub_(1)
+        return xs, ys
+
+
+def fast_collate(batch: List[Tuple[Image.Image, Tensor]], memory_format: torch.memory_format = torch.contiguous_format):
```

Comment:
This code looks _really_ similar to what we have for imagenet. Would it be possible to refactor it into a shared helper class and then that for here and for imagenet? (Same thing for the TransformationFn above)

Reply:
Yeah that would make the most sense, I just tried to avoid it 😅 What do you mean by shared helper class? Would it be sufficient to generalize the function to be able to handle both 1D and 2D targets? Would this be placed in `composer/utils/data.py`?

Reply:
I generalized the ADE20k and ImageNet versions and moved them to `utils/data.py`. The first few epochs of of both datasets seem to align with previous runs. I'm not convinced my code was the best, so feel free to provide critiques.

Reply:
Yup looks good. Thanks!